controllers/kafkauser_controller.go

```diff
@@ -198,6 +198,12 @@ func (r *KafkaUserReconciler) Reconcile(ctx context.Context, request reconcile.R
 	var kafkaUser string
 
 	if instance.Spec.GetIfCertShouldBeCreated() {
+		// Validate the KafkaUser instance annotations before creating a certificate request
+		err := instance.Spec.ValidateAnnotations()
+		if err != nil {
+			return requeueWithError(reqLogger, "failed to reconcile kafkauser", err)
```

Comment:
**Question**: Will this error be wrapped somewhere for context specific information? It may be improved upon if not.

```suggestion
			return requeueWithError(reqLogger, "failed to reconcile kafkauser while validating annotations", err)
```